univocity-trader-core/src/main/java/com/univocity/trader/indicators/MeanDeviationIndicator.java

```diff
@@ -0,0 +1,65 @@
+package com.univocity.trader.indicators;
+
+import com.univocity.trader.candles.Candle;
+import com.univocity.trader.indicators.base.MultiValueIndicator;
+import com.univocity.trader.indicators.base.TimeInterval;
+import com.univocity.trader.strategy.Indicator;
+
+import java.util.function.ToDoubleFunction;
+
+public class MeanDeviationIndicator extends MultiValueIndicator {
+
+    private double value;
+    private final MovingAverage sma;
+
+    public MeanDeviationIndicator(TimeInterval interval) {
+        this(5, interval);
+    }
+
+    public MeanDeviationIndicator(int length, TimeInterval interval) {
+        this(length, interval, null);
+    }
+
+    public MeanDeviationIndicator(int length, TimeInterval interval, ToDoubleFunction<Candle> valueGetter) {
+        super(length, interval, valueGetter == null ? c -> c.close : valueGetter);
+        this.sma = new MovingAverage(length, interval);
+    }
+
+    @Override
+    protected boolean calculateIndicatorValue(Candle candle, double value, boolean updating) {
+
+        if (sma.accumulate(candle)) {
+
+            int index = values.size() - 1;
+
+            double absoluteDeviations = 0;
+
+            final double average = sma.getValue();
+            final int startIndex = Math.max(0, index - 5 + 1);
+            final int nbValues = index - startIndex + 1;
+
+            for (int i = startIndex; i <= index; i++) {
```

Comment:
don't iterate on values of a circular list like this. You need to start from `i = values.getStartingIndex()`, then increment `i` using `i = (i + 1) % values.size();` to make `i` get back to the beginning of the array when needed. Do that `n` times, where `n is the result of values.size()`.